File: parley/src/main/java/nu/parley/android/data/repository/EventRepository.java
package nu.parley.android.data.repository;

import nu.parley.android.data.net.Connectivity;
import nu.parley.android.data.net.service.EventService;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public interface EventRepository {

    public void fire(String event);
}

File: parley/src/main/java/nu/parley/android/data/repository/EventRepository.kt
package nu.parley.android.data.repository

import nu.parley.android.Parley
import nu.parley.android.data.net.ParleyHttpRequestMethod

internal class EventRepository {
    fun fire(event: String?) {
        val network = Parley.getInstance().network
        network.networkSession.request(
            network.url + network.path + "services/event/$event",
            null,
            ParleyHttpRequestMethod.Post,
            onCompletion = {
                // Ignore
            },
            onFailed = { _, _ ->
                // Ignore
            }
        )
    }
}
